### ISSUE TITLE
TM-T88III - Strange issue with dithered image

### ISSUE THREAD

**Issue by DOCaCola** [mike42/escpos-php#1092]
I experience a strange issue when trying to print more "complex" images. I have attached two image files, one is a dithered version and one is a non-dithered version. The non-dithered version seems to print fine every time, while the dithered version is distorted. I have attached photos of the result.

![IMG_4105](https://user-images.githubusercontent.com/2077396/126910929-b7ec42d7-8c05-4ec6-909c-6e25f9d8a250.jpeg)
![IMG_4106](https://user-images.githubusercontent.com/2077396/126910936-1cc8a75d-3b71-4943-afcc-292010897e6c.jpeg)

My setup: I have a TM-T88III connected to a raspberry pi via usb-serial adapter. The graphics library being used should be imagemagick. I have to use bitImage() since graphics() doesn't seem to be working.

$connector = new FilePrintConnector("/dev/serial/by-id/usb-067b_2303-if00-port0");
$profile = CapabilityProfile::load("TM-T88III");    
$printer = new Printer($connector, $profile);
$printer -> initialize();
$tux = EscposImage::load('dithered.png', false);
$printer -> bitImage($tux, Printer::IMG_DOUBLE_WIDTH | Printer::IMG_DOUBLE_HEIGHT);
$printer -> feed();
$printer -> cut();
$printer -> close();

![notdithered](https://user-images.githubusercontent.com/2077396/126910807-8185f461-5086-4792-8e41-66d85c0b24af.png)
![dithered](https://user-images.githubusercontent.com/2077396/126910811-d6f777a1-1aab-40cb-a509-db45cccf4608.png)

**Comment by DOCaCola** [comment]
I did some more testing. Using the 'native' image library results in the exact same issue. I also cropped the dithered tux image at the top and the same split in the head appears as when i print the uncropped version.

![ditheredcropped](https://user-images.githubusercontent.com/2077396/126916204-8ef62d4a-bd7b-414b-80b5-722746ec43f0.png)

**Comment by DOCaCola** [comment]
I have output the data generated by escpos-php to a file. What is being sent to the printer appears to be correct. My TM-T88III has a native serial port connected via an usb to rs232 adapter. I have tried changing the printer speed between 19200 and 9600 with no difference to what is being printed. Either the printer doesn't interpret the data right or it is some kind of strange issue with the RS232 adapter (it uses a prolific chipset).

<img width="302" alt="Hex Workshop -  Z: private tmp printoutput txt  2021-07-26 22-49-03" src="https://user-images.githubusercontent.com/2077396/127057220-3a3eaa4a-f716-4032-9dd7-94ce8dc239f9.png">

**Comment by DOCaCola** [comment]
I tested with bitImageColumnFormat. Even there, there are issues when instructing EscposImage to generate a dithered image.

$tux = EscposImage::load('tux.png', false);
$printer -> bitImageColumnFormat($tux);
prints fine

$tux = EscposImage::load('tux.png', true);
$printer -> bitImageColumnFormat($tux);
garbled output
![IMG_4108](https://user-images.githubusercontent.com/2077396/127136886-e393e62f-ff7b-49bd-9570-de83aedb349a.jpeg)
This is the tux.png i used as input:
![Tux](https://user-images.githubusercontent.com/2077396/127136965-d3e2ee11-47cd-4341-bfe1-74c8b039a464.png)

**Comment by DOCaCola** [comment]
I produced an image with 2 pixels height that still give me the pixel shift issue. I will attach it to this post along with the raw output from escpos-php.
![pixeltest](https://user-images.githubusercontent.com/2077396/127291371-2ba121b5-7fed-4715-9f16-dbb02a11f7ab.jpg)
[Archiv.zip](https://github.com/mike42/escpos-php/files/6891515/Archiv.zip)

**Comment by Devetec** [comment]
Did you ever figure out how to fix this?

**Comment by DOCaCola** [comment]
Not really. I have upgraded to a TM-T88V which supports the "graphics" function which works unlike "bitImage" function.
Maybe in bitImage mode the data needs some additional escaping when certain bitmap sequences are sent to the printer. The printer might interpret some of these bitmap sequences as some other command.
Probably looking at the output of the old official windows drivers would be worth a shot for this. Comparing what they send over rs232 compared to escpos-php
